pkg/controller/wavefrontproxy/wavefrontproxy_controller.go

```diff
@@ -217,3 +222,30 @@ func (r *ReconcileWavefrontProxy) reconcileProxySvc(ip *InternalWavefrontProxy,
 
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileWavefrontProxy) reconcilePvc(ip *InternalWavefrontProxy, reqLogger logr.Logger) (reconcile.Result, error) {
```

Comment:
In case the pvc is deleted manually, what is the observed behavior of the Proxy and the expected behavior of the operator ?

In case creating a Pvc is required only during creating a new Proxy then this can be moved to deploy.go as a "func getPvc()" .

Reply:
Currently this is creating PVC using default storage class and size, reconcilePVC can be enhanced to change these two properties in advanced use cases.

Reply:
ok, from what i understood, this function can be moved to deploy.go 
have a func getPvc(ip) -> The function can create a new PVC if not exists, or return the retrieved PVC if exists. 
In future, we can add support to create a new PVC based on new input CR spec params.

Any reconcile* function inside controller should semantically mean this controller is taking care of the lifecycle of this kubernetes resource. Ex: Stuff like automatically recreating a Pvc is it gets deleted etc.

Reply:
Done.